src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -266,7 +267,19 @@ protected override void OnScriptFileChanged(object sender, FileSystemEventArgs e
                     {
                         TraceLevel level = (TraceLevel)logData["lvl"];
                         var evt = new TraceEvent(level, message);
-                        traceWriter.Trace(evt);
+
+                        // Node captures the AsyncLocal value of the first invocation, which means that logs
+                        // are correlated incorrectly. Here we'll overwrite that value with the correct value
+                        // immediately before logging.
+                        using (logger.BeginScope(
+                            new Dictionary<string, object>
+                            {
+                                ["MS_FunctionInvocationId"] = invocationContext.ExecutionContext.InvocationId
+                            }))
+                        {
+                            // TraceWriter already logs to ILogger
+                            traceWriter.Trace(evt);
```

Comment:
For logging within functions, we'll wire up an ILogger if one is registered: https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Bindings/TraceWriter/TraceWriterBinding.cs#L54-L59

This allows anyone currently using TraceWriter to automatically get App Insights logs if they enable them.